x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -53,9 +55,10 @@ type Dimension struct {
 
 // Config holds a configuration specific for cloudwatch metricset.
 type Config struct {
-	Namespace  string      `config:"namespace" validate:"nonzero,required"`
-	MetricName string      `config:"metricname"`
-	Dimensions []Dimension `config:"dimensions"`
+	Namespace          string      `config:"namespace" validate:"nonzero,required"`
+	MetricName         string      `config:"metricname"`
+	Dimensions         []Dimension `config:"dimensions"`
+	ResourceTypeFilter string      `config:"resource_type_filter"`
```

Comment:
As for [the comment by Carlos](https://github.com/elastic/beats/pull/12480#issuecomment-501352636),  what about doing this mandatory and remove `getResourceTypeUsingNamespace`, at least by now?
```suggestion
	ResourceTypeFilter string      `config:"resource_type_filter" validate:"required"`
```
We can later reintroduce some more complete implementation and make this setting optional, what wouldn't be a breaking change.

Reply:
I just realized after seeing this comment that in some cases this metricset can be used to collect custom metrics, also maybe services without tags? With that in mind maybe keeping it as optional makes more sense, but disabling automatic guessing entirely. Tag enrichment would only be done if `resource_type_filter` is given.

Reply:
> Tag enrichment would only be done if `resource_type_filter` is given.

+1 to that. Should we place tag-related options under a different namespace (as `tags.resource_type_filter`) so it is more explicit that they are for tag collection?

Reply:
OK sounds great! I am more than glad to remove magic functions like `getResourceTypeUsingNamespace`. I like `tags.resource_type_filter`, if it exists, then start collecting tags. If not, ignore the tags.

Reply:
Done, @jsoriano @exekias this is ready for review again  😬 Thank you for all the suggestions!!